docs/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.md

```diff
@@ -89,7 +89,7 @@ The options in Table 2 can be categorized into two types:
 1) options that cause the compiler to augment the produced binary with run-time checks aimed to detect memory errors, and
 2) options that direct the compiler to adjust the properties of the generated binary or code to ensure the resulting binary is compatible with OS-enforced protection mechanisms.
 
-As these options have an impact on the binary produced by the compiler the effects of enabling any of the options in Table 2 must be validated through testing.
+As these options have an impact on the binary produced by the compiler the effects of enabling any of the options in Table 2 must be validated through testing. Many of these compiler options can have context-specific performance impact that in general use is negligible. The detailed descriptions of these options in this document elaborate on in which context non-neglible performance impact may manifest itself.
```

Comment:
Maybe something like:

"Security options can sometimes have a context-specific performance impact that in general use, is either negligible or the benefits far outweigh the overhead. The default descriptions of these options in this document elaborate on in which context non-neglible performance impact may manifest itself. If performance is a concern, then it is recommended to run benahcmarks for your application to understand the actual nature of the slowdown and weigh the benefits against the security protections."